test-refresh/src/com/jakemccrary/test_refresh.clj

```diff
@@ -118,21 +119,26 @@
                      selectors)]
      ns)))
 
-(defn run-selected-tests [test-paths selectors]
+(defn run-selected-tests [test-paths selectors report]
   (let [test-namespaces (namespaces-in-directories test-paths)
         selected-test-namespaces (nses-selectors-match selectors test-namespaces)]
-    (binding [clojure.test/report capture-report]
-      (reset! failed-tests #{})
-      (summary
-       (suppress-unselected-tests selected-test-namespaces
-                                  selectors
-                                  #(apply clojure.test/run-tests selected-test-namespaces))))))
-
-(defn- run-tests [test-paths selectors]
+    (if report (require (symbol (namespace (symbol report)))))
+    (let [resolved-report (when report (let [rr (resolve (symbol report))]
+                                         (if rr rr (println "Unable to locate report method:" report))
+                                         ))
+          used-report (if resolved-report resolved-report capture-report)]
```

Comment:
Actually, could we extract lines 125-129 to a function called something like `select-reporting-fn`